web/server/go.md

```diff
@@ -0,0 +1,498 @@
+### Models
+
+Put simply, models describe data.
+They are agnostic to the data source
+(JSON over HTTP, in-memory data structures, DB tables, etc.),
+however in many cases the term will be used synonymously
+with data queried from a DB.
+
+The model code should provide the structure of the data
+along with methods dictating how outside sources interact with it.
+
+For example,
+here's a pretty simple example of a model struct for a dog.
+
+```go
+package models
+
+type Dog struct {
+    ID      uuid.UUID `json:"id" db:"id"`
+    Name    string    `json:"name" db:"name"`
+    Owner     Owner
+}
+```
+
+It has a a unique ID, name, and `Owner`.
+The latter being a `struct` that encompasses an association.
+
+The field tags follow the syntax patterns for the
+[json](https://golang.org/pkg/encoding/json/)
+and [sqlx](https://godoc.org/github.com/jmoiron/sqlx) packages.
+This is not to prescribe specific packages,
+but to provide a more decorative example that may be encountered in a Truss codebase.
+
+Interacting with this model is likely going to happen in two ways:
+
+1. marshalling from an outside data source
+2. modifying/viewing the data in Go
+   (ex. executing business logic).
+
+#### Marshalling Models
+
+In the most basic cases of accepting an HTTP request
+or writing/reading from a SQL DB,
+marshalling is unlikely to happen manually.
+Many web app frameworks or DB connectors/ORMs
+are going to handle that hidden by their public APIs
+and provide convenience query methods
+or, in the case of JSON HTTP requests,
+parameter wrappers for handlers.
+Within a handler or service,
+the model is likely to be accessed only in its Go struct form.
+If you're curious what that marshalling code might look like,
+you can look at this [json blog post](https://blog.golang.org/json-and-go)
+or take a look at some of the
+[sqlx column mapping code](https://github.com/jmoiron/sqlx/blob/master/sqlx.go#L23)
+that works in tandem with the sql standard library.
+We'll get more into the interaction between those packages
+and our application code in the next section about [services](#services).
+
+#### Working with Models in Go
+
+In the second case,
+we need to utilize our models to store results of our application logic.
+After all, model structs and databases don't deliver client value on their own.
+Let's take a look back at our dog model
+and perform some operations.
+
+First, let's try to add a dog.
+
+```go
+package models
+
+type Dog struct {
+    ID      uuid.UUID `json:"id" db:"id"`
+    Name    string    `json:"name" db:"name"`
+    Owner     Owner
+}
+```
+
+Elsewhere in the code,
+we create a new dog and save it.
+
+```go
+package services
+
+func MyDogFunc(name string, owner Owner) (Dog, error) {
+    dog := Dog{}
+    dog.Name = name
+    dog.Owner = Owner
+    // Save is an external method to interface with DB passed as dependency
+    db.Save(&dog)
+    return dog, nil
+}
+```
+
+Notice how our data source is queried entirely outside of the model.
```

Comment:
This is one of the areas of "I'm being opinionated". I think large models caused some trouble in the MilMove codebase, and I'm pretty firm on using a different pattern. That being said, this was only tested in a few models and hasn't been tested throughout an application.